### PR TITLE
bpo-44987: Speed up unicode normalization of ASCII strings

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue44987

### CODE REVIEW

**Comment by corona10** [comment]
```
>>> from timeit import Timer
>>> setup="from unicodedata import normalize; s = 'reverse'"
>>> t1 = Timer('normalize("NFKC", s)', setup=setup)
>>> setup="from unicodedata import normalize; s = 'reverse'*1000"
>>> t2 = Timer('normalize("NFKC", s)', setup=setup)
>>> min(t1.repeat(repeat=7))
0.038022067994461395
>>> min(t2.repeat(repeat=7))
0.038196470006369054
```

File: Doc/whatsnew/3.11.rst [hunk 1]
```diff
@@ -233,6 +233,13 @@ sqlite3
   :issue:`16379`.)
 
 
+unicodedata
+-----------
+* If the given string is pure ASCII string, :func:`unicode.normalize` now
+  handles this as an already normalized to process it in constant time.
+  (Contributed by Dong-hee Na in :issue:`bpo-44987`.)
```
Comment (serhiy-storchaka) [inline Doc/whatsnew/3.11.rst]: It is not worth a new subsection in the "Improved Modules" section. An entry in the "Optimizations" section should be enough. And make it shorter. E.g. "Pure ASCII strings are now normalized in constant time."

File: Misc/NEWS.d/next/Library/2021-09-11-14-41-02.bpo-44987.Mt8DiX.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+If the given string is pure ASCII string, :func:`unicode.normalize` now
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Library/2021-09-11-14-41-02.bpo-44987.Mt8DiX.rst]: unicodedata

File: Doc/whatsnew/3.11.rst (hunk 1 above)
Comment (corona10) [reply Doc/whatsnew/3.11.rst]: Thanks :) Nice suggestion

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by vstinner** [comment]
Nice optimization, thanks @corona10!

By the way, I'm not sure why ".pdbrc is now read with utf-8 encoding." is mentioned in the Optimization section.

**Comment by vstinner** [comment]
> By the way, I'm not sure why ".pdbrc is now read with utf-8 encoding." is mentioned in the Optimization section.

I created PR #28518 for that.

**Pull request merged**